options/options.go

```diff
@@ -1142,8 +1164,9 @@ func (interp *interpreter) interpretField(mc *internal.MessageContext, msg proto
 						mc, ood.Name(), fieldName(existingFld))
 				}
 			}
-			fdm = dynamicpb.NewMessage(fld.Message())
-			msg.Set(fld, protoreflect.ValueOfMessage(fdm))
+			fldVal := msg.NewField(fld)
+			fdm = fldVal.Message()
+			msg.Set(fld, fldVal)
```

Comment:
Oof, this is a real issue in the compiler today. It's never been tickled because in all prior versions of protobuf, there are no options that are messages: only custom options could be messages, so it was safe to use dynamic messages. But with the usage of `google.protobuf.FeatureSet` in the tests, this blew up because line 1146 was trying to set the field using a dynamic message, but `msg` was not dynamic and expected a proper `*descriptorpb.FeatureSet` instance.